trie/snapshotTrieStorageManager.go

```diff
@@ -47,7 +51,11 @@ func (stsm *snapshotTrieStorageManager) GetFromOldEpochsWithoutAddingToCache(key
 
 // PutInEpochWithoutCache saves the data in the given epoch storer
 func (stsm *snapshotTrieStorageManager) PutInEpochWithoutCache(key []byte, data []byte, epoch uint32) error {
-	if stsm.checkIfClosed(fmt.Sprintf("putInEpochWithoutCache context closing, key = %s", hex.EncodeToString(key))) {
+	stsm.tsm.storageOperationMutex.Lock()
```

Comment:
this is very dangerous: you locked the mutex from the tsm to be sure the closed is consistent but you call the pruningStorer.PutInEpochWithoutCache which is another component that locks another mutex there. The code is prone to deadlocks.

Reply:
Refactored as discussed.